modules/user_accounts/php/NDB_Form_user_accounts.class.inc

```diff
@@ -300,80 +330,71 @@ class NDB_Form_User_Accounts extends NDB_Form
                         array('fullName' => $values['Real_name'])
                     );
                 } else {
+
                     $DB->update(
                         'examiners',
                         array(
-                         'radiologist' => $values['examiner_radiologist'] ?? 0,
+                         'radiologist' => $ex_radiologist,
                          'userID'      => $uid,
                         ),
                         array(
                          "full_name" => $values['Real_name'],
                         )
                     );
                     $examinerID = $examinerID[0]['examinerID'];
-                }
 
-                // Update CenterIDs
-                $ex_curr_sites = array();
-                $ex_prev_sites = array();
-
-                $prev_sites = $DB->pselect(
-                    "SELECT centerID
+                    //get existing sites for examiner
+                    $prev_sites = $DB->pselect(
+                        "SELECT centerID
                      FROM examiners_psc_rel epr
                      WHERE examinerID=:eid",
-                    array("eid" => $examinerID)
-                );
+                        array("eid" => $examinerID)
+                    );
 
-                //get sites where user is already an examiner at for compare
-                foreach ($prev_sites as $row => $center) {
-                    array_push($ex_prev_sites, $center['centerID']);
+                    //get sites where user is already an examiner at for compare
+                    foreach ($prev_sites as $row => $center) {
+                        array_push($ex_prev_sites, $center['centerID']);
+                    }
                 }
 
-                foreach ($values as $k => $v) {
-                    //examiner fields
-                    if (preg_match("/^ex_[0-9]+$/", $k)) {
```

Comment:
pregmatch was moved to the top, used to check for the need to manipulate examiner data.

only array with the sites is passed here